### PR TITLE
Add the time_will_tell gem back, using the I18n.with_locale method

### PR DESCRIPTION
**Pull request by jfly**
This is an alternative fix for #1128.

See:
 - #1133
 - #1128
 - mbillard/time_will_tell#5

### CODE REVIEW

**Comment by jfly** [comment]
@viroulep, could you take a look, as I'm attempting to undo some of your changes here?

File: WcaOnRails/app/views/competitions/_admin_index_table.html.erb [hunk 1]
```diff
@@ -33,13 +33,13 @@
           <tr>
             <td>
               <i class="flag f16 <%= competition.country.iso2.downcase %>"></i><%= link_to competition.cellName, competition_path(competition) %>
-              <br /><strong><%= competition.country.name_in(:en) %></strong>, <%= competition.cityName %>
```
Comment (viroulep) [inline WcaOnRails/app/views/competitions/_admin_index_table.html.erb]: Is this intentional?

I'm fine with it but it looks more like a "localize admin page" than a "restore time_will_tell" thing :)

**Comment by jfly** [comment]
> The way the diff is displayed for the indentation changes is a bit messy

Yes it is! GitHub pro tip: add the `w=1` url parameter to any page with a diff to ignore whitespace. For example, on this PR: https://github.com/thewca/worldcubeassociation.org/pull/1172/files?w=1.

File: WcaOnRails/app/views/competitions/_admin_index_table.html.erb (hunk 1 above)
Comment (jfly) [inline WcaOnRails/app/views/competitions/_admin_index_table.html.erb]: Whoops! This was so long ago, I don't know what I was thinking... if anything. Will revert my changes to this file.

File: WcaOnRails/app/views/delegate_reports/_delegate_report.html.erb [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
   <div>
     <strong>Location</strong>
-    <%= competition.cityName %>, <%= competition.country.name_in(:en) %>
+    <%= competition.cityName %>, <%= competition.country.name %>
```
Comment (viroulep) [inline WcaOnRails/app/views/delegate_reports/_delegate_report.html.erb]: I didn't this one the first time :(
That's the same idea: this page is not localized yet, so when it's rendered outside a "with_locale" the country name will be localized, which I think doesn't belong to this PR.

Comment (jfly) [inline WcaOnRails/app/views/delegate_reports/_delegate_report.html.erb]: Good call, will revert my changes to this file.
